packages/ens-controller/src/EnsController.ts

```diff
@@ -295,6 +277,38 @@ export class EnsController extends BaseController<
     return true;
   }
 
+  #setDefaultEthProvider(registriesByChainId?: Record<number, Hex>) {
+    const { selectedNetworkClientId } = this.messagingSystem.call(
+      'NetworkController:getState',
+    );
+    this.#setEthProvider(selectedNetworkClientId, registriesByChainId);
+  }
+
+  #setEthProvider(
+    selectedNetworkClientId: string,
+    registriesByChainId?: Record<number, Hex>,
+  ) {
+    const selectedNetworkClient = this.messagingSystem.call(
+      'NetworkController:getNetworkClientById',
+      selectedNetworkClientId,
+    );
+    const currentChainId = selectedNetworkClient.configuration.chainId;
+
+    if (
+      registriesByChainId &&
+      registriesByChainId[parseInt(currentChainId, 16)] &&
+      this.#getChainEnsSupport(currentChainId)
+    ) {
+      this.#ethProvider = new Web3Provider(selectedNetworkClient.provider, {
+        chainId: convertHexToDecimal(currentChainId),
+        name: CHAIN_ID_TO_ETHERS_NETWORK_NAME_MAP[currentChainId as ChainId],
+        ensAddress: registriesByChainId[parseInt(currentChainId, 16)],
+      });
+    } else {
+      this.#ethProvider = null;
+    }
+  }
+
```

Comment:
Do you think that using [`NetworkController:getSelectedNetworkClient`](https://github.com/MetaMask/core/blob/8ffcb839b1f3183316bcbbe26375a1c7bd7458d8/packages/network-controller/src/NetworkController.ts#L301) would simplify the logic here letting NetworkController do the hard work, or is it unusable here for some reason?

Reply:
This comment probably applies also to the `constructor` logic

Reply:
Ah sorry, I see, we also need the `clientId` which is not provided by `:getSelectedNetworkClient`. 
Though, do we need to create a new instance of `Web3Provider`? is there a reason why we can't use what's provided by `NetworkController` in this case? 

I see that in the current extension implementation the `provider` passed to `EnsController` comes directly from `NetworkController`: https://github.com/MetaMask/metamask-extension/blob/cd30d782a3097a00983601ab3211261d8b63b207/app/scripts/metamask-controller.js#L530